metrics/web/docs/features/deep_links/diagrams/updating_page_parameters_sequence_diagram.puml

```diff
@@ -0,0 +1,24 @@
+@startuml updating_page_parameters_sequence_diagram
+skinparam ParticipantPadding 20
+skinparam BoxPadding 10
+
+participant MetricsRouterDelegate << (C,#ADD1B2) >>
+participant NavigationNotifier << (C,#ADD1B2) >>
+participant PageParametersFactory << (C,#ADD1B2) >>
+
+activate MetricsRouterDelegate
+MetricsRouterDelegate -> NavigationNotifier : handleNewRoutePath(routeConfiguration)
+activate NavigationNotifier
+deactivate MetricsRouterDelegate
+
+NavigationNotifier --> NavigationNotifier : processes route configuration
+
+NavigationNotifier -> PageParametersFactory : create(routeConfiguration)
+activate PageParametersFactory
+
+PageParametersFactory --> NavigationNotifier : coolPageParametersModel
+deactivate PageParametersFactory
+
+NavigationNotifier --> NavigationNotifier : sets new page parameters
```

Comment:
:readable:
Should we also notify listeners here? As they could be any I suggest naming the participant as `Presentation` or something like this.